control-plane/agents/src/bin/core/node/specs.rs

```diff
@@ -169,4 +170,19 @@ impl ResourceSpecsLocked {
         registry.store_obj(&drained_node_spec).await?;
         Ok(drained_node_spec)
     }
+
+    pub(crate) async fn set_node_drained(
+        &self,
+        registry: &Registry,
+        node_id: &NodeId,
+    ) -> Result<NodeSpec, SvcError> {
+        let node = self.get_locked_node(node_id)?;
```

Comment:
IIRC we previously discussed adding a ResourceMutex and OperationGuard for nodes, guess that can come after?

Reply:
As discussed let's drop this for now.